legend-engine-xt-persistence-test-runner/src/main/java/org/finos/legend/engine/testable/persistence/extension/PersistenceTestH2Connection.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.testable.persistence.extension;
+
+import org.eclipse.collections.impl.factory.Lists;
+import org.finos.legend.engine.authentication.credential.CredentialSupplier;
+import org.finos.legend.engine.authentication.demoflows.H2LocalWithDefaultUserPasswordFlow;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.DatasetDefinition;
+import org.finos.legend.engine.persistence.components.relational.jdbc.JdbcHelper;
+import org.finos.legend.engine.plan.execution.stores.relational.connection.authentication.strategy.TestDatabaseAuthenticationStrategy;
+import org.finos.legend.engine.plan.execution.stores.relational.connection.driver.vendors.h2.H2Manager;
+import org.finos.legend.engine.plan.execution.stores.relational.connection.ds.specifications.LocalH2DataSourceSpecification;
+import org.finos.legend.engine.shared.core.identity.Identity;
+import org.finos.legend.engine.shared.core.identity.factory.IdentityFactoryProvider;
+
+import javax.security.auth.Subject;
+import java.sql.Connection;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+public class PersistenceTestH2Connection
+{
+    private Connection connection;
+
+    Connection getConnection()
+    {
+        // Close connection to close any left over open connections
+        closeConnection();
+        Properties properties = new Properties();
+        properties.put("DATABASE_TO_UPPER", false);
+        LocalH2DataSourceSpecification specification = new LocalH2DataSourceSpecification(
+            Lists.mutable.empty(),
+            new H2Manager(),
+            new TestDatabaseAuthenticationStrategy(),
+            properties);
+        Identity identity = IdentityFactoryProvider.getInstance().makeIdentity((Subject) null);
+        this.connection = specification.getConnectionUsingIdentity(identity, plainTextCredentialSupplier());
```

Comment:
Can we take a look at how this null identity shows up in the engine logs ? Today we have a problem where a ton of relational activity is logged as the __UNKNOWN__ user and it makes it hard for the SRE teams to figure out what's going on. Are we able to access the "real current user " here ? If yes, maybe we should log the user as "alice_fortest" or something like that ?

Reply:
@chloeminkyung and @hardikmaheshwari Presumably want all test activity in engine logs to follow the same convention, so we can filter out?